docs/rules/no-reject-literal.md

```diff
@@ -0,0 +1,57 @@
+# disallow using literals as Promise rejection reasons (no-reject-literal)
+
+It is considered good practice to only instances of the built-in `Error` object for user-defined errors in Promises. `Error` objects automatically store a stack trace, which can be used to debug an error by determining where it came from. If a Promise is rejected with a non-`Error` value, it can be difficult to determine where the rejection occurred.
+
+
+## Rule Details
+
+This rule aims to ensure that Promises are only rejected with `Error` objects.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint no-reject-literal: "error"*/
+
+Promise.reject("something bad happened");
+
+Promise.reject(5);
+
+Promise.reject();
```

Comment:
There are many parallels between promises and syntactic try/catch, indeed. Promises, however, are *more* powerful than that, and don't have to be limited to try/catch. What's the `Promise.all` or `Promise.race` parallel in try/catch?